src/stat.c

```diff
@@ -134,29 +140,61 @@ static inline void StatUpdateMinMax(uint64_t newValue, uint64_t *min, uint64_t *
     *min = (newValue < *min) ? newValue : *min;
 }
 
+static void resetStats(struct Statistics *stats) {
+    memset(stats, 0, sizeof(struct Statistics));
+    stats->RoundTripMin = UINT64_MAX;
+}
+
 void StatFrameReceived(enum StatFrameType frameType, uint64_t cycleNumber, bool outOfOrder, bool payloadMismatch,
                        bool frameIdMismatch)
 {
     struct RoundTripContext *rtt = &RoundTripContexts[frameType];
     struct Statistics *stat = &GlobalStatistics[frameType];
+    struct Statistics *statPerPeriodPre = &GlobalStatisticsPerPeriodPrep[frameType];
+
     struct timespec rxTime = {};
     uint64_t rtTime;
+    uint64_t currTime, elapsedT;
+
 
     LogMessage(LOG_LEVEL_DEBUG, "%s: frame[%" PRIu64 "] received\n", StatFrameTypeToString(frameType), cycleNumber);
 
     /* Record Rx timestamp in us */
     if (LogRtt)
     {
         clock_gettime(appConfig.ApplicationClockId, &rxTime);
-        rtTime = TsToNs(&rxTime) - rtt->Backlog[cycleNumber % rtt->BacklogLen];
+        currTime = TsToNs(&rxTime);
+        rtTime = currTime - rtt->Backlog[cycleNumber % rtt->BacklogLen];
         rtTime /= 1000;
 
+        if (statPerPeriodPre->FirstTimeStamp == 0)
+            statPerPeriodPre->FirstTimeStamp = currTime;
+
+        /* Test if the amount of time specified in the config is arrived.
+         * if true this will be the last point to be taken into stats per period */
+        elapsedT = currTime - statPerPeriodPre->FirstTimeStamp;
+        if (elapsedT >= appConfig.StatsCollectionIntervalNS)
+        {
+            statPerPeriodPre->ready = true;
+            statPerPeriodPre->LastTimeStamp = currTime;
+        }
+
         StatUpdateMinMax(rtTime, &stat->RoundTripMin, &stat->RoundTripMax);
+        StatUpdateMinMax(rtTime, &statPerPeriodPre->RoundTripMin, &statPerPeriodPre->RoundTripMax);
+
         if (StatFrameTypeIsRealTime(frameType) && rtTime > RttExpectedRTLimit)
+        {
             stat->RoundTripOutliers++;
+            statPerPeriodPre->RoundTripOutliers++;
+        }
         stat->RoundTripCount++;
+        statPerPeriodPre->RoundTripCount++;
+
         stat->RoundTripSum += rtTime;
+        statPerPeriodPre->RoundTripSum += rtTime;
+
         stat->RoundTripAvg = stat->RoundTripSum / (double)stat->RoundTripCount;
+        statPerPeriodPre->RoundTripAvg = statPerPeriodPre->RoundTripSum / (double)statPerPeriodPre->RoundTripCount;
```

Comment:
Is it possible to guard the whole `statPerPeriodPre` updates with `#if WITH_MQTT`, e.g., by moving it into a separate function?

Reply:
Makes sense as well!